Makefile

```diff
@@ -12,21 +12,43 @@ IMAGE_TAG ?= latest
 OPERATOR_IMAGE ?= cluster-network-addons-operator
 REGISTRY_IMAGE ?= cluster-network-addons-registry
 
-vet:
+cmd_sources := $(shell find ./cmd/ -type f -name '*.go')
+pkg_sources := $(shell find ./pkg/ -type f -name '*.go')
+apis_sources := $(shell find ./pkg/apis/ -type f -name '*.go')
```

Comment:
Good we have to be carefull if we delete files, since timestamp will not change.

Reply:
Have learn yesterday a pair of tricks we can use here
- First of all we can compose a custom with make functions (it's faster since they run directly in the running make memory)

```make
# Make does not offer a recursive wildcard function, so here's one:
rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
```

- Then we can use it like 
```make
cmd_sources := $(call rwildcard,cmd/,*.go)
```

- Also to truck files renames or deleted we just add the directory (it also re-build if subdirs have changed)
```make
cmd_sources := $(call rwildcard,cmd/,*.go) cmd/
```

- So all together would be:
```make
# Make does not offer a recursive wildcard function, so here's one:
rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))

# Gather needed source files and directories to create target dependencies
cmd_sources := $(call rwildcard,cmd/,*.go) cmd/
pkg_sources := $(call rwildcard,pkg/,*.go) pkg/
apis_sources := $(call rwildcard,pkg/apis/,*.go) pkg/apis/
```

With all this we can maybe face false rebuilds but not false noops.